spec/models/order_cycle_spec.rb

```diff
@@ -551,6 +551,30 @@
     end
   end
 
+  describe "opened_at " do
+    let!(:oc) {
+      create(:simple_order_cycle, orders_open_at: 2.days.ago, orders_close_at: 1.day.ago, opened_at: 1.week.ago)
+    }
+
+    it "reset opened_at if open date change in future" do
+      expect(oc.opened_at).to_not be_nil
+      oc.update!(orders_open_at: 1.week.from_now, orders_close_at: 2.weeks.from_now)
+      expect(oc.opened_at).to be_nil
```

Comment:
I don't think we enforce a limit on the number of `expect` per example (maybe it's something to look at), but I like to try to stick to one if possible.
I think it would be better to set the value of `oc.opened_at` instead of checking if it's not nil. What do you think ?

Reply:
That's probably a good idea, at least for unit tests. For integration/system specs, I think it's necessary to have multiple expectations.

In this case, I was checking a pre-condition, but the better way to do that would be with `change`. I'll update.

Reply:
Updated in new commit.

Reply:
> That's probably a good idea, at least for unit tests. For integration/system specs, I think it's necessary to have multiple expectations.
True, I forgot about system specs